### PR TITLE
Support <svg:image crossorigin>

### PR DESCRIPTION
**Pull request by annevk**
#### 504cb3567c094a4118abbe6812443cdd431fb374
<pre>
Support &lt;svg:image crossorigin&gt;
<a href="https://bugs.webkit.org/show_bug.cgi?id=260724">https://bugs.webkit.org/show_bug.cgi?id=260724</a>
rdar://114462749

Reviewed by Youenn Fablet.

Add support for the crossorigin attribute to SVGImageElement in a
manner analogous to HTMLImageElement.

Also remove a duplicate parser for the crossorigin attribute in
HTMLImageElement.

* LayoutTests/imported/w3c/web-platform-tests/svg/idlharness.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.crossOriginSource.sub.html:
* Source/WebCore/html/HTMLImageElement.cpp:
(WebCore::HTMLImageElement::attributeChanged):
(): Deleted.
(WebCore::parseCrossoriginState): Deleted.
* Source/WebCore/svg/SVGImageElement.cpp:
(WebCore::SVGImageElement::attributeChanged):
(WebCore::SVGImageElement::setCrossOrigin):
(WebCore::SVGImageElement::crossOrigin const):
* Source/WebCore/svg/SVGImageElement.h:
* Source/WebCore/svg/SVGImageElement.idl:

Canonical link: <a href="https://commits.webkit.org/267349@main">https://commits.webkit.org/267349@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/3c628a28904c290b790f9ddec2a14077658ba630

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16284 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16603 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17022 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18052 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15271 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16472 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19683 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16716 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17651 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16478 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16918 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13939 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18819 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14163 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14740 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21559 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15150 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14905 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/18214 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15498 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13134 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14718 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/14581 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3914 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19083 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15325 "Built successfully") | | |

### CODE REVIEW

File: LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.crossOriginSource.sub.html [hunk 1]
```diff
@@ -1,222 +1,219 @@
...
+      if ('requestVideoFrameCallback' in video)
+        video.requestVideoFrameCallback(callback);
+      else
+        video.onloadeddata = callback;
+    }
 
-</script>
+  </script>
 </body>
+
```
Comment (youennf) [inline LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.crossOriginSource.sub.html:218]: Are these changes needed?
It seems related to tabs/whitespaces and I do not see any -expected.txt change

**Review by youennf — APPROVED** [review]

File: LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.crossOriginSource.sub.html (hunk 1 above)
Comment (annevk) [reply LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-construction.crossOriginSource.sub.html:218]: Yeah, I imported this test anew from upstream and without the changes I make here we would end up failing this test: https://wpt.fyi/results/webcodecs/videoFrame-construction.crossOriginSource.sub.html.

https://github.com/web-platform-tests/wpt/commit/9ff836132efe33400736a4c1fe0c76bd4e6808ab

This is the test that is being changed:

> Test creating a VideoFrame with a CORS enabled cross-origin SVGImageElement with crossorigin="anonymous"

Expected does not change because the test still has the same name and is passing after this change.
